pandas/_libs/tslibs/period.pyx

```diff
@@ -886,7 +886,10 @@ cdef int64_t get_time_nanos(int freq, int64_t unix_date, int64_t ordinal) nogil:
         # We must have freq == FR_HR
         factor = 10**9 * 3600
 
-    sub = ordinal - unix_date * 24 * 3600 * 10**9 / factor
```

Comment:
so the trouble is that there is an overflow going on somewhere in this expression?

Reply:
Yea I am fairly certain that `24 * 3600 * 10**9` will overflow - these are likely interpreted by the compiler to just be of type int, but that multiplication could very well exceed the limits of an int type. Adding the ULL suffix I think would be ideal 

More details on how decimal literals are assigned types here:
https://stackoverflow.com/a/41407498/621736

Reply:
Using uint64_t instead of int64_t will work for the given example, but will then fail for date range earlier than the epoch with an integer overflow, so we must stick with signed integer here.
About how this change work, it just change the order of the operation so that unix_date is not multiplied by `24 * 3600 * 10**9`, but by `24 * 3600 * 10**9 / factor`, which is smaller and does not result into an integer overflow (except for value in the really far futur for the use case described in the issue, after the year `2*10**15`)
So the real fix to do here is maybe just to add parenthesis in the right place, see new commit shortly.

Reply:
```
In [64]: 24 * 3600 * 10**9                                                                                                                                
Out[64]: 86400000000000

In [65]: np.iinfo(np.int64).max                                                                                                                           
Out[65]: 9223372036854775807

```

this is a fairly standard number, i agree if multiplied by a large number this *could* overflow, but ok here.